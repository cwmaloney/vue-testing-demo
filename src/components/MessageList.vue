<template>
  <div class="message-list">
    <h1>Messages</h1>
    <ul>
      <li v-for="message in messages" :key="message.id">
        {{ message.id }}: {{ message.message }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'message-list',
  props: [
    'messages'
  ]
}
</script>

<style lang="css" scoped>
.message-list {
  border: solid;
  margin: 10px 0;
  padding: 5px;
  }
</style>
