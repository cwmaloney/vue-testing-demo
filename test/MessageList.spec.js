import { shallowMount } from '@vue/test-utils'
import MessageList from '../src/components/MessageList.vue'

describe('MessageList.vue', () => {
  it('renders li for each message', () => {
    const messages = ['Apple', 'Orange', 'Banana'];
    const wrapper = shallowMount(MessageList, {
      propsData: { messages }
    });
    expect(wrapper.findAll('li')).toHaveLength(messages.length);
  });
});
