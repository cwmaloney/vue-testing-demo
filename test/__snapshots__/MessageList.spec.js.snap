// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList.vue rendors items 1`] = `
<div class="message-list">
  <h1>Messages</h1>
  <ul></ul>
</div>
`;
